src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { CSSTransition, TransitionGroup } from 'react-transition-group';
+import { Todo } from '../../types/Todo';
+import { TodoItem } from '../TodoItem/TodoItem';
+
+interface Props {
+  todos: Todo[];
+  tempTodo: Todo | null;
+  isDeletingCompleted: boolean;
+  isUpdatingAllTodo: boolean;
+  toggleStatus: boolean;
+}
+
+export const TodoList: React.FC<Props> = React.memo(({
+  todos,
+  tempTodo,
+  isDeletingCompleted,
+  isUpdatingAllTodo,
+  toggleStatus,
+}) => {
+  return (
+    <TransitionGroup>
+      {todos.map((todo) => (
+        <CSSTransition
+          key={todo.id}
+          timeout={300}
+          classNames="item"
+        >
+          <TodoItem
+            todo={todo}
+            isPerentLoading={
+              (isDeletingCompleted && todo.completed)
+              || (isUpdatingAllTodo && todo.completed === toggleStatus)
+            }
+          />
+        </CSSTransition>
+      ))}
+
+      {tempTodo !== null && (
+        <CSSTransition
+          key={0}
+          timeout={300}
+          classNames="temp-item"
+        >
+          <TodoItem
+            todo={tempTodo}
+            isPerentLoading={tempTodo !== null}
```

Comment:
The condition 'tempTodo !== null' is always true at this point, as it is already checked in line 39. You can change it to 'true' or remove the prop if it's not necessary.